pkg/reconciler/reconciler.go

```diff
@@ -651,7 +650,9 @@ func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (res ctrl.
 			return ctrl.Result{}, err
 		}
 	default:
-		return ctrl.Result{}, fmt.Errorf("unexpected release state: %s", state)
+		err := fmt.Errorf("unexpected release state: %s", state)
```

Comment:
Not sure if this is in scope, but I just realized the inconsistency of whether we set or redefine `err` (`err :=` rather than `err =`) here, in the preceding `case` as well as in the prehook loop might have unintended consequences, given that it affects the [defer block](https://github.com/operator-framework/helm-operator-plugins/pull/226/files#diff-fc2bc6354d4506ced4adfd45dda2737dc917c217e282041d9af5bce57e514a85R559) for the updater Apply call.

I wonder whether we should take a closer look at this.